### PR TITLE
trigger decryption via manual pii audit data request

### PR DESCRIPTION
**Pull request by zaydgds**

### CODE REVIEW

File: src/sharedServices/sns/publishToSNS.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SNSClient, PublishCommand } from '@aws-sdk/client-sns'
```
Comment (gdsandrewmccafferty) [inline src/sharedServices/sns/publishToSNS.ts:1]: Can we add some tests for this? Apart from anything else, it'll push our test coverage down if we don't, and also every other similar bit of code that interacts with the AWS SDK currently has tests.

File: src/lambdas/initiateAthenaQuery/handler.test.ts [hunk 2]
```diff
@@ -174,4 +185,18 @@ describe('initiate athena query handler', () => {
...
+    mockPublishToSNS.mockResolvedValue('messageID')
+    await handler(testManualAthenaQueryEvent, mockLambdaContext)
+    expect(mockPublishToSNS).toHaveBeenCalledWith(
+      'arn:aws:sns:eu-west-2:123456789012:email-to-slack-topic',
+      `Retrieved data for zendeskID: ${testZendeskId}`
+    )
+    expect(logger.info).toHaveBeenCalledWith(
+      'Manual query detected, no need to run athena query'
+    )
+    expect(confirmAthenaTable).not.toHaveBeenCalled()
```
Comment (gdsandrewmccafferty) [inline src/lambdas/initiateAthenaQuery/handler.test.ts:200]: If you're leaving the other dependencies in place, I think I'd like to see some assertions that they're also not called. At a minimum, you want to make sure `startQueryExecution` isn't called.

File: src/sharedServices/sns/publishToSNS.ts (hunk 1 above)
Comment (zaydgds) [reply src/sharedServices/sns/publishToSNS.ts:1]: added a test

File: src/lambdas/initiateAthenaQuery/handler.test.ts (hunk 2 above)
Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/handler.test.ts:200]: added a wrapper

File: src/lambdas/initiateAthenaQuery/handler.test.ts [hunk 3]
```diff
@@ -1,21 +1,6 @@
...
-import { getDatabaseEntryByZendeskId } from '../../sharedServices/dynamoDB/dynamoDBGet'
-import { createQuerySql } from './createQuerySql'
-import { updateQueryByZendeskId } from '../../sharedServices/dynamoDB/dynamoDBUpdate'
-import { ConfirmAthenaTableResult } from '../../types/athena/confirmAthenaTableResult'
-import {
-  dataPathsTestDataRequest,
-  noIdTestDataRequest,
-  testDataRequest
-} from '../../utils/tests/testDataRequest'
+import * as initiateQueryImportWrapper from './initiateQuery'
```
Comment (gdsandrewmccafferty) [inline src/lambdas/initiateAthenaQuery/handler.test.ts:2]: You don't need this any more, or the spy

File: src/lambdas/initiateAthenaQuery/handler.test.ts [hunk 4]
```diff
@@ -42,152 +27,8 @@ jest.mock('../../sharedServices/sns/publishToSNS', () => ({
...
-    expect(mockUpdateZendeskTicket).toHaveBeenCalledWith(
-      testZendeskId,
-      'test athena error',
-      'closed'
-    )
-    expect(mockUpdateQueryByZendeskId).not.toHaveBeenCalled()
-  })
-})
-
 describe('tests related to running manual queries', () => {
```
Comment (gdsandrewmccafferty) [inline src/lambdas/initiateAthenaQuery/handler.test.ts:32]: You need a test for the "normal" scenario, where now you'll be asserting that the parameters from the handler are passed through correctly. You also need to reinstate some of the tests that are to do with the event body being malformed.

File: src/lambdas/initiateAthenaQuery/handler.ts [hunk 5]
```diff
@@ -22,95 +16,14 @@ export const handler = async (
 
   const zendeskId = retrieveZendeskIdFromEvent(event)
```
Comment (gdsandrewmccafferty) [inline src/lambdas/initiateAthenaQuery/handler.ts:17]: I would put retrieveZendeskIdFromEvent back in it the handler, and then put the tests for the event structure being malformed here. It doesn't make sense to do that in the other file, and then export it.

File: src/lambdas/initiateAthenaQuery/initiateQuery.test.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+  Promise<ConfirmAthenaTableResult>
+>
+const mockUpdateZendeskTicket = updateZendeskTicketById as jest.Mock
+const mockGetDatabaseEntryByZendeskId = getDatabaseEntryByZendeskId as jest.Mock
+const mockCreateQuerySql = createQuerySql as jest.Mock
+const mockUpdateQueryByZendeskId = updateQueryByZendeskId as jest.Mock
+const mockStartQueryExecution = startQueryExecution as jest.Mock
+const mockPublishToSNS = publishToSNS as jest.Mock
+
+describe('initiate athena query handler', () => {
```
[the remark this answers is not in the document]
Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/initiateQuery.test.ts:55]: done

File: src/lambdas/initiateAthenaQuery/handler.test.ts (hunk 3 above)
Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/handler.test.ts:2]: done

File: src/lambdas/initiateAthenaQuery/handler.test.ts (hunk 4 above)
Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/handler.test.ts:32]: done

File: src/lambdas/initiateAthenaQuery/handler.ts (hunk 5 above)
Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/handler.ts:17]: done

Comment (zaydgds) [reply src/lambdas/initiateAthenaQuery/handler.ts:17]: done

**Pull request merged**